src/ImageOverlayDialog.cc

```diff
@@ -38,6 +38,23 @@ ImageOverlayDialog::ImageOverlayDialog(QPixmap image, QWidget *parent)
 	raise();
 }
 
+void ImageOverlayDialog::reject()
+{
+	// needed on macOS to recover the main menu after the dialog is closed(!)
+	setWindowState(Qt::WindowNoState);
+
+	// It might be slightly cleaner to do:
+	//
+	// #ifdef Q_WS_MACX
+	// #include <Carbon/Carbon.h>
```

Comment:
The solution is good enough for me. Since I'm pretty new to Qt myself, I might have missing something or the full screen widget isn't implemented properly. So maybe it's not a Qt bug but we have missed the proper API calls.

You can remove `raise()`, it's probably a leftover from previous iterations.